src/Simplify_Call.cpp

```diff
@@ -506,6 +506,38 @@ Expr Simplify::visit(const Call *op, ExprInfo *bounds) {
         } else {
             return Call::make(op->type, op->name, {arg}, op->call_type);
         }
+    } else if (op->is_intrinsic(Call::if_then_else)) {
+        // Note that this call promises to evaluate exactly one of the conditions,
+        // so this optimization should be safe.
+
+        internal_assert(op->args.size() == 3);
+        Expr cond_value = mutate(op->args[0], nullptr);
+        Expr true_value = mutate(op->args[1], nullptr);
+        Expr false_value = mutate(op->args[2], nullptr);
+
+        // Ignore likelies for our purposes here
+        Expr cond = cond_value;
+        if (const Call *c = cond.as<Call>()) {
+            if (c->is_intrinsic(Call::likely) ||
+                op->is_intrinsic(Call::likely_if_innermost)) {
+                cond = c->args[0];
+            }
+        }
+
+        if (is_one(cond)) {
+            return true_value;
```

Comment:
I propose not mutating true_value and false_value above, and instead returning mutate(true_value, bounds), so that the bounds can be tracked.

Reply:
Will do, but on that note: is the proper usage of the bounds parameter adequately documented for the Simplifier code? It wasn't immediately obvious to me when I should propagate bounds and when I should not.

Reply:
PTAL

Reply:
bounds is just an out-param passed by pointer, so you should only pass the bounds pointer directly into the mutation of that Expr if you're going to return that mutated Expr.